src/osd/osd_types.cc

```diff
@@ -5826,9 +5827,10 @@ void SnapSet::encode(ceph::buffer::list& bl) const
 
 void SnapSet::decode(ceph::buffer::list::const_iterator& bl)
 {
-  DECODE_START_LEGACY_COMPAT_LEN(3, 2, 2, bl);
+  DECODE_START_LEGACY_COMPAT_LEN(4, 2, 2, bl);
   decode(seq, bl);
   bl += 1u;  // skip legacy head_exists (always true)
+  std::vector<snapid_t> snaps;  // removed in 4
```

Comment:
See above.  It wasn't actually removed in 4.

Reply:
marking as dnm until resolved